src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -186,15 +186,26 @@ PaymentSandbox::balanceHook (AccountID const& account,
     {
         auto delta = amount.zeroed ();
         auto lastBal = amount;
+        auto minBal = amount;
         for (auto curSB = this; curSB; curSB = curSB->ps_)
         {
             if (auto adj = curSB->tab_.adjustments (account, issuer, currency))
             {
                 delta += adj->debits;
                 lastBal = adj->origBalance;
```

Comment:
Thanks for the explanation.  I see it now.  The `minBal` is the smallest starting balance seen for this `(account, issuer, currency)` in any of the recursive `PaymentSandboxes`.  So you're finding the smallest post-computation balance regardless of the depth of PaymentSandboxes.  That looks right.

You might consider adding your explanation as a comment in the code.  Personally, I didn't know enough about `PaymentSandbox` internals to get what you were up to just by reading the code.